src/spine.js

```diff
@@ -127,15 +85,37 @@ pc.extend(pc, function () {
 
         this._position = new pc.Vec3();
 
-        var atlas = new spine.TextureAtlas(atlasData, function (path) {
-            return new SpineTextureWrapper(textureData[path]);
-        });
+        var atlas;
+
+        // EP: As instructed: https://github.com/playcanvas/playcanvas-spine/pull/73
+        //
+        // API differs before we can know which which file version this is supposed to support.
+        // The data file opened will determine which code paths are used.  This file may
+        // or may not be concatenated with the spine library that supports it.
+        //
+        // use the length of the function signatures to guess which library has been concatenated.
+        if (spine.TextureAtlas.length === 1) {
+            // spine 4.1
+            atlas = new spine.TextureAtlas(atlasData);
+            for (const page of atlas.pages) {
+                page.setTexture(new SpineTextureWrapper(textureData[page.name]));
+            }
+        } else {
+            // spine 3.6 and 3.8
+            atlas = new spine.TextureAtlas(atlasData, function (path) {
+                return new SpineTextureWrapper(textureData[path]);
+            });
+        }
+
         var json = new spine.SkeletonJson(new spine.AtlasAttachmentLoader(atlas));
         json.scale *= 0.01;
         var _skeletonData = json.readSkeletonData(skeletonData);
-        this.skeletonVersion = _skeletonData.version;
-        this._spine_3_6_0 = versionCompare(this.skeletonVersion, "3.6.0") <= 0;
-        this._spine_3_7_99 = versionCompare(this.skeletonVersion, "3.7.99") <= 0;
+
+        // Compatibility queries
+        this.skeletonVersion = semver.valid(semver.coerce(_skeletonData.version));
+        this._spine_3_6_0 = semver.satisfies(this.skeletonVersion, '<3.6.0'); // version 3.6.0 or below
```

Comment:
![image](https://github.com/playcanvas/playcanvas-spine/assets/697563/b513cb25-b7d9-4d8b-8c6d-655d4f2dc838)

Reply:
My comment is wrong.  It should be "below version 3.6.0"

The value used is tested on the behaviour of the function that was used before.

Reply:
But doesn't `versionCompare(this.skeletonVersion, "3.6.0") <= 0` say "less than or equal to 3.6.0"?

Reply:
You're right.  Ran a test with:

```
  console.log(`'3.5.0' ${versionCompare("3.5.0", "3.6.0") <= 0}`);
  console.log(`'3.6.0' ${versionCompare("3.6.0", "3.6.0") <= 0}`);
  console.log(`'3.6.1' ${versionCompare("3.6.1", "3.6.0") <= 0}`);
```

Updating now.

Reply:
completed in 29b6250